packages/material/src/util/debounce.ts

```diff
@@ -0,0 +1,16 @@
+import { debounce } from 'lodash';
+import { useState, useCallback } from 'react'
+
+
+const eventToValue = (ev: any) => ev.target.value;
+export const useDebouncedChange = (handleChange: (path: string, value: any) => void, defaultValue: any, data: any, path: string, eventToValueFunction: (ev: any) => any = eventToValue, timeout = 300): [any, React.ChangeEventHandler, () => void] => {
+  const [input, setInput] = useState(data || defaultValue);
+  const debouncedUpdate = useCallback(debounce((newValue: string) => handleChange(path, newValue), timeout), [handleChange]);
```

Comment:
This should be a `useMemo` as we are calculating a function and not creating the function ourselves in place. A proper ESLint setup would complain here ;) 
```suggestion
  const debouncedUpdate = useMemo(() => debounce((newValue: string) => handleChange(path, newValue), timeout), [handleChange]);
```

Reply:
we are actually creating a function, a function that is used to call handleChange

Reply:
Yes that's true, and technically that's completely fine. 

I just noticed the pattern because ESLint usually complaints about it ("exhaustive-deps") as it can't properly analyze the function and therefore doesn't give you the usual guarantees. As a workaround we can use `useMemo`. We just don't have ESLint yet configured. I'm fine leaving it like this, so we can also change this later when we finally use ESLint.

Reply:
Now that we're talking about it: `timeout` is missing in the dependencies as well as `path`.